Low: controld: Update for OCF 1.1

nrwahl2 opened the pull request.

kgaillot reviewed (commented):
Looks good so far. Also need to update the validate command to do the checks other than globally-unique only on level 10 (since they only apply to the node running the resource).

nrwahl2 commented:
> do the checks other than globally-unique only on level 10

Updated the OCF 1.1 commit accordingly. Also added one more commit on top to make the globally-unique check a bit more robust.

kgaillot commented on extra/resources/controld.in:
```diff
@@ -64,20 +64,23 @@ In most cases, it should be run as an anonymous clone.
+<content type="select" default="dlm_controld">
+  <option value="gfs_controld" />
+  <option value="dlm_controld" />
+</content>
```
Looking at this again, I don't think we restricted the daemon name before -- these were just the possible defaults

Theoretically someone could have a custom daemon under a different name

nrwahl2 replied:
Good catch

kgaillot commented on extra/resources/controld.in:
```diff
@@ -260,17 +263,19 @@ controld_monitor() {
+    if [ "$OCF_CHECK_LEVEL" = "10" ]; then
+        check_binary killall
+        check_binary "${OCF_RESKEY_daemon}"
+        [ -d "@runstatedir@/cluster" ] || mkdir "@runstatedir@/cluster"
```
Ouch, I just realized this was a little hacky before -- not only did it validate, but it created this directory if needed. start/monitor call validate (without a check level). So we still need to do this, but it doesn't belong in the validate function

nrwahl2 replied:
Ahhh yep.

kgaillot reviewed (commented):
Noticed a couple more things

nrwahl2 commented:
Replaced last two commits to address review.
* Reverted `daemon` content type to `string` since a user could specify a custom daemon if they wanted.
* Moved `@runstatedir@/cluster` check and creation from `controld_validate()` to `controld_start()`.

nrwahl2 commented:
The OCF 1.1 update probably should have had "Feature" in the header. May want to write it down somewhere for Changelog inclusion.
